src/Caching/Storages/FileStorage.php

```diff
@@ -143,7 +143,7 @@ public function lock($key)
 	{
 		$cacheFile = $this->getCacheFile($key);
 		if ($this->useDirs && !is_dir($dir = dirname($cacheFile))) {
-			@mkdir($dir); // @ - directory may already exist
+			@mkdir($dir, 0777, true); // @ - directory may already exist
```

Comment:
@milo thanx!

Btw, FileStorage is using only one level of directories, isn't it? So this PR is not needed.